src/main/java/eu/xenit/alfred/content/gateway/collections/ObservableMap.java

```diff
@@ -0,0 +1,122 @@
+package eu.xenit.alfred.content.gateway.collections;
+
+import static reactor.core.publisher.Sinks.EmitFailureHandler.FAIL_FAST;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import lombok.Value;
+import lombok.experimental.Delegate;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+
+
+public class ObservableMap<K, V> implements Map<K, V>, AutoCloseable {
+
+    @Delegate(types = MapReadOperations.class)
+    private Map<K, V> map = new LinkedHashMap<>();
+    private final Sinks.Many<MapUpdate<K, V>> updates = Sinks.many().multicast().onBackpressureBuffer();
+
+
+    @Override
+    public V get(Object key) {
+        return this.map.get(key);
```

Comment:
Can also be part of the `MapReadOperations`

Reply:
Actually, it cannot be part of `MapReadOperations` due to generics shenanigans:

The class passed to `@Delegate` cannot have generic type parameters, so you end up with this compile error:

```
> Task :compileJava FAILED
contentgrid-gateway/src/main/java/eu/xenit/alfred/content/gateway/collections/ObservableMap.java:15: error: ObservableMap is not abstract and does not override abstract method get(Object) in Map
public class ObservableMap<K, V> implements Map<K, V>, AutoCloseable {
       ^
  where V is a type-variable:
    V extends Object declared in class ObservableMap
/home/tgeens/xenit/contentgrid/contentgrid-gateway/src/main/java/eu/xenit/alfred/content/gateway/collections/ObservableMap.java:17: error: get(Object) in ObservableMap cannot implement get(Object) in Map
    @Delegate(types = MapReadOperations.class)
    ^
  return type Object is not compatible with V#2
  where V#1,V#2 are type-variables:
    V#1 extends Object declared in interface Map
    V#2 extends Object declared in class ObservableMap
```